cli/standalone.rs

```diff
@@ -150,6 +150,10 @@ pub async fn create_standalone_binary(
     } else {
       output
     };
+  if output.exists() {
+    let error_msg = format!("File / Directory {:?} already exists. Please specify another output name.", &output);
```

Comment:
If a file with the inferred name exists, should it throw this error or just overwrite it? If this error is thrown, we would have to delete the compiled file every time before each compile.

Reply:
I think we need to consider the common pattern of making a change and updating the bundle...  No other tool complains when you are updating the output, it overwrites by default.  The only problem is when the output is a directory, or trying to overwrite an input file...

Reply:
Maybe if we put some sort of signature in the output file, we might be able to detect whether it's an output file that can be overwritten or an input file, which means need to throw error. Just random idea.

Reply:
Or we can use flag `--overwrite` that allows the CLI to overwrite the file if the file exists.

Reply:
> Or we can use flag --overwrite that allows the CLI to overwrite the file if the file exists.

I think that would be too much.

> Maybe if we put some sort of signature in the output file, we might be able to detect whether it's an output file that can be overwritten or an input file, which means need to throw error. Just random idea.

Or could be some magic number like 0xcafebabe in Java.

Reply:
There is already a magic signature in the last 16 bytes. We could use that.

Reply:
The compile command now check for the magic trailer and will only overwrite when it exists. I think this design is good.

But commonly, compiler would overwrite whatever file is the output (e.g, `tsc`, `rustc`, `javac`). So I kinda doubt whether we should check for this and prefer to simply overwrite the output file if it's exists (which is the current implementation).

Reply:
tsc, rustc, javac or gcc's output files by default are with an extension, like .js, .so, .class, and .out. The default filename of deno compile however is the source file without extension, so it's more likely to collide with other files or directories. So I think it makes sense to do this check. If it's not too complicated.